### ISSUE TITLE
Speed up AbstractRow.equals() and hashCode() when comparing with other AbstractRow

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#14747]
Interestingly, `AbstractRow` and its many subtypes don't override `AbstractQueryPart::hashCode` and `AbstractQueryPart::equals`, so the default applies, which is never good (see https://github.com/jOOQ/jOOQ/issues/11126):

<img width="714" alt="image" src="https://user-images.githubusercontent.com/734593/222471092-5de15f17-b9c4-4f21-838e-044f9acc6c10.png">

Let's see what a benchmark yields.

**Comment by lukaseder** [comment]
Curiously, `FieldsImpl` does override the methods, since #6804

**Comment by lukaseder** [comment]
**Before**

```
Benchmark                     Mode  Cnt        Score      Error  Units
EqualsHashCode.equalsRow     thrpt    7   789568.901 ┬▒ 1049.829  ops/s
EqualsHashCode.hashCodeRow   thrpt    7  1552855.673 ┬▒ 3953.645  ops/s
EqualsHashCode.notEqualsRow  thrpt    7   775447.660 ┬▒ 1521.801  ops/s
```

**After**:

```
Benchmark                     Mode  Cnt         Score         Error  Units
EqualsHashCode.equalsRow     thrpt    7  78137407.459 ┬▒ 3762165.599  ops/s
EqualsHashCode.hashCodeRow   thrpt    7  92974042.394 ┬▒  827003.424  ops/s
EqualsHashCode.notEqualsRow  thrpt    7  73434078.381 ┬▒ 1917341.050  ops/s
```

Clearly, a huge win.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0, 3.17.9 (https://github.com/jOOQ/jOOQ/issues/14748), and 3.16.15 (https://github.com/jOOQ/jOOQ/issues/14749)
